### PR TITLE
Remove Thrust patch in favor of CMake definition for Thrust 32-bit offset types.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Follow-up for #17523 to use `target_compile_definitions` and drop the Thrust patch.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by vyasr — APPROVED** [review]

File: cpp/cmake/thirdparty/get_cccl.cmake [hunk 1]
```diff
@@ -36,6 +36,12 @@ function(find_and_configure_cccl)
   # Find or install CCCL with our custom set of patches
   rapids_cpm_cccl(BUILD_EXPORT_SET cudf-exports INSTALL_EXPORT_SET cudf-exports)
 
+  # Remove this after upgrading to a CCCL that has a proper CMake option. See
+  # https://github.com/NVIDIA/cccl/pull/2844
+  target_compile_definitions(
+    cudf PRIVATE THRUST_FORCE_64_BIT_OFFSET_TYPE=0 THRUST_FORCE_32_BIT_OFFSET_TYPE=1
+  )
+
```
Comment (vyasr) [inline cpp/cmake/thirdparty/get_cccl.cmake:44]: Sorry I didn't check where you put this. This needs to go into cpp/CMakeLists.txt after the cudf target has been created.

Comment (bdice) [reply cpp/cmake/thirdparty/get_cccl.cmake]: Fixed in 816fc09.

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -928,6 +928,12 @@ if(TARGET CUDA::cuFile${_cufile_suffix})
   target_compile_definitions(cudf PRIVATE CUDF_CUFILE_FOUND)
 endif()
 
+# Remove this after upgrading to a CCCL that has a proper CMake option. See
+# https://github.com/NVIDIA/cccl/pull/2844
+target_compile_definitions(
+  cudf PRIVATE THRUST_FORCE_64_BIT_OFFSET_TYPE=0 THRUST_FORCE_32_BIT_OFFSET_TYPE=1
```
Comment (bdice) [inline cpp/CMakeLists.txt:934]: We got an error:
```
#  error "Only THRUST_FORCE_32_BIT_OFFSET_TYPE or THRUST_FORCE_64_BIT_OFFSET_TYPE may be defined!"
```
so I will remove the 64 bit definition.
```suggestion
  cudf PRIVATE THRUST_FORCE_32_BIT_OFFSET_TYPE=1
```

**Pull request merged**